packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1493,17 +1511,6 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     }
 
     private isContainerMessageDirtyable(type: ContainerMessageType, contents: any) {
```

Comment:
This method would need to be overridden somehow when using makeLegacyContainerRuntime, right?
I think it will be hard to deal with all of this without having derived class.

Reply:
I don't think we need to do anything special here.  The idea of this change is that `AgentScheduler` just becomes a normal data store, and so the `ContainerRuntime` has no reason to treat it specially.

Reply:
Sure, but I assume this breaks Bohemia scenarios where they want to exclude ops for AgentScheduler when calculating such things as
- dirty document flag
- last edited field for doc
I.e. you are right at this layer, but how do we help them make existing scenarios working the way they were working before for their (legacy) container?

Reply:
The real answer is that the runtime shouldn't own "last edited" determination since it has no way of knowing which messages are semantically relevant to an end user's perception of what it means to edit the document.  Any DataObject could be using ops in a manner that a user wouldn't equate to an "edit" in an end-user's mind (e.g. "commit"-like messages from Tablero), so I don't see this being a sustainable solution even beyond the `AgentScheduler`.

I'll see if I can come up with some ideas to mitigate in the short term but I don't think a long-term solution looks like overriding this method.

Reply:
Looks like there's adequate opportunity to exclude _scheduler messages from last-edit consideration in `shouldDiscardMessage()`, added a note to BREAKING.md.